src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509Encoder.cs

```diff
@@ -174,7 +174,10 @@ public unsafe void DecodeX509KeyUsageExtension(byte[] encoded, out X509KeyUsageF
                 if (data->pathlen != IntPtr.Zero)
                 {
                     hasPathLengthConstraint = true;
-                    pathLengthConstraint = Interop.libcrypto.ASN1_INTEGER_get(data->pathlen).ToInt32();
+                    long pathLengthConstraint64 = Interop.Crypto.Asn1IntegerGet(data->pathlen);
+
+                    Debug.Assert(pathLengthConstraint64 < int.MaxValue, "pathLengthConstraint needs to be in the Int32 range.");
```

Comment:
It won't throw. We have unchecked everywhere. I want to make debug checked, but there's a lot of test failures when I tried it. If we want to throw always, we should use explicit 'checked'.